packages/apps-engine/src/server/managers/AppListenerManager.ts

```diff
@@ -1058,6 +1064,22 @@ export class AppListenerManager {
     }
 
     // Livechat
+    private async executePreLivechatRoomCreatePrevent(data: ILivechatRoom): Promise<boolean> {
+        let prevented = false;
+
+        for (const appId of this.listeners.get(AppInterface.IPreLivechatRoomCreatePrevent)) {
+            const app = this.manager.getOneById(appId);
+
+            prevented = (await app.call(AppMethod.EXECUTE_PRE_LIVECHAT_ROOM_CREATE_PREVENT, data)) as boolean;
+
+            if (prevented) {
+                return prevented;
+            }
+        }
+
+        return prevented;
```

Comment:
although this is very similar to other code doing same stuff, I'd suggest simplify it to something like:

```suggestion
        for (const appId of this.listeners.get(AppInterface.IPreLivechatRoomCreatePrevent)) {
            const app = this.manager.getOneById(appId);

            if (await app.call(AppMethod.EXECUTE_PRE_LIVECHAT_ROOM_CREATE_PREVENT, data)) {
                return true;
            }
        }

        return false;
```